File: server/management/src/main/java/dev/irmlerjo/management/order/order.java
package dev.irmlerjo.management.order;

import java.time.LocalDateTime;

import com.fasterxml.jackson.annotation.JsonFormat;

import dev.irmlerjo.management.costreport.FixedPriceCostReport;
import dev.irmlerjo.management.costreport.PerHourCostReport;

public record Order(String id, String projectId, String name, String description,
		FixedPriceCostReport fixedPriceCostReport, PerHourCostReport perHourCostReport,
		@JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm:ss") LocalDateTime deliveryDate,
		Integer plannedHourlyBudget) {

	public Order(String id, Order order) {
		this(id, order.projectId(), order.name(), order.description(), order.fixedPriceCostReport(),
				order.perHourCostReport(), order.deliveryDate(), order.plannedHourlyBudget());
	}
}

File: client/management-client/src/app/model/order.ts
import { FixedPriceCostReport } from "./fixedpricecostreport";
import { PerHourCostReport } from "./perhourcostreport";

export interface Order {
    id: string,
    projectId:string,
    name:string,
    description:string,
    fixedPriceCostReport:FixedPriceCostReport,
    perHourCostReport:PerHourCostReport,
	deliveryDate:Date,
    plannedHourlyBudget:number
    currentHours?:number
}